### ISSUE TITLE
Workers on mon_mitm are using same routes

### ISSUE THREAD

**Issue by kamieniarz** [Map-A-Droid/MAD#679]
This issue was pretty rare for a long time (or maybe it was fixed one day?) but recently I'm seeing it very often.

The issue: workers using mon_mitm are using same routes instead of "their own" ones. Example: I have 4 workers in the same area with 200 positions, each worker gets 50 positions but one of them walks on the same route as second worker, so part of the area isn't scanned at all. It basically looks like on the screenshot (walker following walker)
![mad](https://user-images.githubusercontent.com/9063907/72290758-3acbb880-364e-11ea-83dd-9c9d3e624d6a.png)

It's trully annoying and the only way to fix it is reloading mad by pressing `Apply settings` or by restarting mad, so it can allocate routes to workers properly.

It happens when one of the workers laggs out and routes needs to be re-allocate to each worker BUT one of the workers kind of "doesn't care" about it and keeps doing what he was doing:
![mad2](https://user-images.githubusercontent.com/9063907/72291518-d1e54000-364f-11ea-9ff3-28530c6b8fa5.png)

Picture from above shows situation when worker 01 joined to the others co route was split again and allocated to other workers (co they started from position 1) but worker 06 ignored that fact and kept doing "his job" so by that one of the other workers was allocated to partially the same route as 06...

I hope I'm clear enough and as I remember when I reported this bug on Discord (few days ago I guess) someone also confirmed to have that issue.

**Comment by HELLspawnTIM** [comment]
Can confirm, same for me. It also happens sometimes when workers change their area.

**Comment by kamieniarz** [comment]
Thanks to @Expl0dingBanana I'm able to prove what I'm saying :D
![mad3](https://user-images.githubusercontent.com/9063907/72302928-d28bcf80-366b-11ea-90f1-86433af861f1.png)
These 2 workers in one route are 01 and 05 and they're both on the same route, leaving one route without any worker. 

I find it very frustrating :/

**Comment by cecpk** [comment]
I need a debug log pls

**Comment by kamieniarz** [comment]
@cecpk I uploaded my logfile: https://www26.zippyshare.com/v/3iwLFr0Y/file.html

So basically I restarted my workers X96m01-X96m05 to force bug. Workers X96m01-X96m11 are in the same area. After these 5 atvs restarted bug appeared - 2 atvs were on the same route (same situation as on the screenshot from previous post). 
Log starts when all 5 atvs start to connect to MAD and ends when I realized bug appeared. 

I hope that's enough
